airflow/decorators/base.py

```diff
@@ -208,11 +208,35 @@ def __init__(
         # since values for those will be provided when the task is run. Since
         # we're not actually running the function, None is good enough here.
         signature = inspect.signature(python_callable)
+
+        # Don't allow context argument defaults other than None to avoid ambiguities.
+        if any(
+            param.name in KNOWN_CONTEXT_KEYS and param.default not in (None, inspect.Parameter.empty)
+            for param in signature.parameters.values()
+        ):
+            raise ValueError("Context key parameters can't have a default other than None")
```

Comment:
Would it be helpful if this shows which argument is at fault? Or is it unnecessary?

Reply:
New error message:

```
ValueError: Context key parameter logical_date can't have a default other than None
```

Definitely helps debugging. We could also make it print all (or a subset) of distinct faulty context key parameters, but this should be enough in my opinion. What do you think?